eclass/systemd.eclass

```diff
@@ -53,20 +53,21 @@ _systemd_get_dir() {
 
 	if $(tc-getPKG_CONFIG) --exists systemd; then
 		d=$($(tc-getPKG_CONFIG) --variable="${variable}" systemd) || die
-		d=${d#${EPREFIX}}
 	else
-		d=${fallback}
+		d="${EPREFIX}${fallback}"
 	fi
 
 	echo "${d}"
 }
 
-# @FUNCTION: _systemd_get_systemunitdir
+# @FUNCTION: _systemd_unprefix
+# @USAGE: <function-name>
 # @INTERNAL
 # @DESCRIPTION:
-# Get unprefixed unitdir.
-_systemd_get_systemunitdir() {
-	_systemd_get_dir systemdsystemunitdir /lib/systemd/system
+# Calls the specified function and removes ${EPREFIX} from the result.
+_systemd_unprefix() {
+	local d=$("${@}")
+	echo "${d#${EPREFIX}}"
```

Comment:
Not a huge deal given it's unlikely `${EPREFIX}` would have glob characters and it was like that before, but note that this syntax does allow globs to work, e.g. for a weirdo using a literal `EPREFIX="/*"; d="/*/lib"`, it would echo `*/lib` rather than `/lib`.
```suggestion
	echo "${d#"${EPREFIX}"}"
```

Reply:
Ah, I never realized that. Thanks for pointing it out.